### PR TITLE
Port light_controlled_intersection_tactical_plugin to ROS2

### PR DESCRIPTION
**Pull request by JonSmet**
# PR Details
## Description

This PR includes updates to port the light_controlled_intersection_tactical_plugin node to ROS2 and to launch it from the guidance.launch.py launch file.

Note that a future JIRA Story (and carma-platform PR) will be created in order to improve the unit test coverage for this package.

## Related Issue

Issue #1889 

## How Has This Been Tested?

Full carma-platform image built, and locally integration tested using the development carma-config.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp [hunk 1]
```diff
@@ -13,417 +13,377 @@
...
+        if(!processed_maneuvers.empty() && processed_maneuvers.back().type == carma_planning_msgs::msg::Maneuver::LANE_FOLLOWING){
+            points_and_target_speeds = add_lanefollow_buffer(wm, points_and_target_speeds, processed_maneuvers, ending_state_before_buffer, detailed_config);
+        }
 
+        return points_and_target_speeds;
+    }
 
-} //namespace light_controlled_intersection_transit_plugin
\ No newline at end of file
+} // light_controlled_intersection_tactical_plugin
```
Comment (kjrush) [inline light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp:389]: The diff for this file is kind of a mess to read, I think git is a bit confused. But as far as i can read it it seems like things are in the right places.

**Review by kjrush — COMMENTED** [review]
Looks good to me, might not hurt to talk about whether or not it's appropriate to use an autoformatting pass on this, as there's a few cases where classes have both "snake_case" and "camelCase" method names, etc. If it's not going to interrupt work on this, I think it'd be a net win.

File: light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin_node.cpp [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  namespace std_ph = std::placeholders;
+
+  LightControlledIntersectionTransitPluginNode::LightControlledIntersectionTransitPluginNode(const rclcpp::NodeOptions &options)
+      : carma_guidance_plugins::TacticalPlugin(options)
+  {
+    // Create initial config
+    config_ = Config();
+
+    // Declare parameters
+    config_.centerline_sampling_spacing = declare_parameter<double>("centerline_sampling_spacing", config_.centerline_sampling_spacing);
```
[the remark this answers is not in the document]
Comment (JonSmet) [reply light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin_node.cpp:29]: Good catch! The match on lines 50-52 would work using the default parameter values defined within `light_controlled_intersection_tactical_plugin_config.hpp`, but I want to apply the multipliers using the loaded configuration parameter values. To fix this, I've moved the math to `on_configure_plugin()` after the parameters have been loaded.

File: carma/launch/guidance.launch.py [hunk 3]
```diff
@@ -76,9 +76,6 @@ def generate_launch_description():
     port_drayage_plugin_param_file = os.path.join(
         get_package_share_directory('port_drayage_plugin'), 'config/parameters.yaml')
     
-    inlanecruising_plugin_file_path = os.path.join(
```
[the remark this answers is not in the document]
Comment (JonSmet) [reply carma/launch/guidance.launch.py:79]: The inlanecruising_plugin configuration file was incorrectly added to guidance.launch.py as part of PR #1843. It was added to both guidance.launch.py and plugins.launch.py, but it was only supposed to be added to plugins.launch.py.

File: light_controlled_intersection_tactical_plugin/test/node_test.cpp [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+#include <chrono>
+#include <thread>
+#include <future>
+
+#include "light_controlled_intersection_tactical_plugin/light_controlled_intersection_tactical_plugin_node.hpp"
+
+namespace light_controlled_intersection_transit_plugin
+{
+    
+    TEST(LCITacticalPluginTest, apply_accel_cruise_decel_speed_profile_test)
```
[the remark this answers is not in the document]
Comment (JonSmet) [reply light_controlled_intersection_tactical_plugin/test/node_test.cpp:30]: Yes correct, I have now added a TODO comment here as well to state that unit tests will be added as part of future work for this package.

File: light_controlled_intersection_tactical_plugin/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# light_controlled_intersection_tactical_plugin
+
+TODO for USER: Add description of package and link to confluence documentation.
```
[the remark this answers is not in the document]
Comment (JonSmet) [reply light_controlled_intersection_tactical_plugin/README.md]: Didn't realize that I'd missed this in my initial commit. Updated.

**Comment by kjrush** [comment]
Can you take a look at the two bugs Sonar reported? They don't appear to be a functional issue, but they do seem like unnecessary loops. Are those present as-is in the original? If they're new, we should definitely fix them, we'll have to consider what to do with them if they're present in the original.

**Review by kjrush — APPROVED** [review]

**Pull request merged**